wandb/wandb_controller.py

```diff
@@ -738,6 +738,47 @@ def _sweep_status(sweep_obj, sweep_conf, sweep_runs):
 
 
 def sweep(sweep, entity=None, project=None):
+    """Initialize a hyperparameter sweep.
+
+        # this line initializes the sweep
+        sweep_id = wandb.sweep({'name': 'my-awesome-sweep',
+                                'program': 'train.py',
+                                'metric': 'accuracy',
+                                'method': 'grid',
+                                'parameters': {'a': {'values': [1, 2, 3, 4]}}})
+
+        # this line actually runs it -- parameters are available to
+        # my_train_func via wandb.config
+        wandb.agent(sweep_id, function=my_train_func)
```

Comment:
We can use the code markdown here like
```python
code here
```
To enclose the entire thing as a code snippet.

We can instead also have an `Examples:` section below where this could go!

Reply:
i moved to an examples section. indentation of 4 spaces is valid markdown for starting a code block, so kept that as is - https://www.markdownguide.org/basic-syntax/#code-blocks-1

Reply:
Due to the fact that we have our references hosted to GitBook, we need to explicitly mention the backticks and the language to have a rich color setting.

Reply:
@dannygoldstein there are some rough edges on our document generation. I'd intended to spare you from all of that and for me and @ariG23498 to figure out the formatting details later, but I didn't discuss that with him.

The style we use for code examples is exemplified in [`wandb.log`](https://github.com/wandb/client/blob/c129c32964aca6a8509d98a0cc3c9bc46f2d8a4c/wandb/sdk/wandb_run.py#L956). Triple backticks and "python".